tests/unit/test_charm.py

```diff
@@ -870,20 +873,14 @@ def test_set_secret(harness):
         # Test application scope.
         assert "password" not in harness.get_relation_data(rel_id, harness.charm.app.name)
         harness.charm.set_secret("app", "password", "test-password")
-        assert (
-            harness.get_relation_data(rel_id, harness.charm.app.name)["password"]
-            == "test-password"
-        )
+        assert harness.charm.get_secret("app", "password") == "test-password"
         harness.charm.set_secret("app", "password", None)
         assert "password" not in harness.get_relation_data(rel_id, harness.charm.app.name)
 
         # Test unit scope.
         assert "password" not in harness.get_relation_data(rel_id, harness.charm.unit.name)
         harness.charm.set_secret("unit", "password", "test-password")
-        assert (
-            harness.get_relation_data(rel_id, harness.charm.unit.name)["password"]
-            == "test-password"
-        )
+        assert harness.charm.get_secret("unit", "password") == "test-password"
```

Comment:
We shouldn't re-write the old test.

We should keep Juju2 compatibilty.

Please add a separate test for secrets, that's only executed when secrets are available

Reply:
I kept the old test, running on juju2 only, as it checks the relation databag for the secret. I kept the changed test running on both versions as it is compatible either way. Left some comments for future perspective. LMK what you think

Reply:
All right, I see, sorry it was my confusion.

Now I understand what you mean, renaming was a good idea here :-)

All good on this one, thank you :-)

Reply:
Actually... My bad sorry, I overlooked this test.

This one is not using secrtes.

IT's only using secrets when secrets are configured but otherwise it's falling back to the databag.

We should have a test that is specifically checking if a Juju secret comes to existence when we create one with `charm.set_secret()`.

I think we had it already somewhere lower down in this file.

We don't need to duplicate that.

I mean: it's good to have a single generic test on set-get secret functionality. It will work both on databag and with Juju Secrets, with no knowledge of the underlying storage.

However we also MUST have specific tests for both databag usage (`@only_without_juju_secrets` -- should fail otherwise) and vica versa.

I think I have some of those in all PG apps `main` right now. That concept has to be kept and followed if we want a single pipeline.